### ISSUE TITLE
Windows installer improvements

### ISSUE THREAD

**Issue by towlie** [qTox/qTox#1607]
Remove require admin by:
- Installing qTox into \AppData\Roaming\qTox
- register Tox URL by using this registry path "HKEY_CURRENT_USER\Software\Classes\Tox"
- uninstall entries should be set into this path "HKEY_CURRENT_USER\Software\Microsoft\Windows\CurrentVersion\Uninstall\qTox

**Comment by towlie** [comment]
Also make sure that your installer doesn't request admin rights by default. Maybe you have to drop it by inserting this into a manifest file:

```
     <trustInfo xmlns="urn:schemas-microsoft-com:asm.v2">
          <security>
              <requestedPrivileges>
                  <requestedExecutionLevel
                      level="asInvoker"
                      uiAccess="false"
                  />
              </requestedPrivileges>
```

Windows tries to run executables as admin if the name contains: setup, installer or update/updater.

**Comment by mikalehmann** [comment]
I would really like to choose where qTox installs to and I don't like AppData for that purpose. Therefore this feature would need an adjustment:
- only when the program doesn't get admin rights will it install to AppData
- if the user grants the installer admin rights they can choose the installation folder

**Comment by towlie** [comment]
This is doable. I remember that a function exists to check if it runs as admin.

**Comment by ghost** [comment]
The thing with the uninstall entries is really important. (it's annoying to not have such standard things) And I think with NSIS (I think I saw somewhere that you use it) it should be possible quite easy.
Another standard thing on Windows is the installation in %programfiles%. I agree to @xur86x - I also don't like this installation in LocalAppData. IMO the pre-selected (default) value for the installation path on Windows should be %programfiles%. (I've also selected this for my installation)

##### As for the uninstallation entry

BTW I noticed just now that there is in fact an uninstallation entry in _HKEY_CURRENT_USER\Software\Microsoft\Windows\CurrentVersion\Uninstall\qTox_. However it's incomplete as there is only one registry key _UninstallString_. At least one entry is missing: _DisplayName_
(And _DisplayIcon_ would be nice too of course)
More information here: https://msdn.microsoft.com/en-us/library/aa372105%28v=vs.85%29.aspx

I think also the key should be changed. At least if the installer has admin rights and it's installed into %programfiles% it should use the key _HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Windows\CurrentVersion\Uninstall_. (especially because tox is not per-user if you install it in %programfiles%)

**Comment by mikalehmann** [comment]
I just want to stress that, although %programfiles% should be the standard installation path, there should be an option to install in every folder I want. Many installers today lack this, because big developers don't care about their users.

**Comment by tux3** [comment]
@blobgo I made a quick fix for the uninstall entry, and properly put it in HKLM by default for now.

**Comment by ghost** [comment]
@xur86x
Yes of course you should be able to choose any folder you want. %programfiles% just should be the default.
E.g. for a portable installation you can then just switch the installation location. (A fancy feature would of course be that - if you choose a removable drive - the installer will detect this an adjust the installation, e.g. by omitting the registry key and setting qTox setting to "make portable" - but only after the user agrees to a question message like "You have chosen a removable drive for installation. Do you want to install qTox in portable mode?". I saw this in another installer and really liked this "intelligence". However this is surely hard to implement and not that important so just keep it in mind for later)

@tux3
Does it even work with non-admin rights in this case? I assume HKLM is only writeable with admin rights.
That's the reason why I said it should choose this based on the install location/admin state.

**Comment by tux3** [comment]
@blobgo It currently only works with admin rights.

The non-adming option isn't implemented at all currently. But that was something I could fix right now.

**Comment by NoxNoctium** [comment]
Why was this closed, most of the proposed changes are either not decided on or implemented  in the installer script yet. [qtox64.nsi](https://github.com/tux3/qTox/blob/master/windows/qtox64.nsi)

**Comment by ghost** [comment]
Actually we don't even get an installer (see https://github.com/tux3/qTox/issues/2085) so I'm also in favour of opening this again.

**Comment by towlie** [comment]
I closed this because I have the feeling that there is no interest in making these changes.

**Comment by zetok** [comment]
@towlie Well, it would be a (very?) low priority, since now installer isn't even used, but it would be definitely a thing that should be done, even if not in any near(?) future.

**Comment by jhasse** [comment]
What do you guys think about a .msi installer? I would be willing to create one, just recently played around with it using WiX.

**Comment by zetok** [comment]
@jhasse is it possible to create them on debian?

**Comment by Zer0-One** [comment]
if you can build it on jenkins, we can serve it from pkg.tox.chat

**Comment by jhasse** [comment]
@zetok WiX is written in C#, so I think it will be difficult. After googling I only found this:http://wixtoolset.org/issues/4381/ So maybe it's possible using Wine?

**Comment by zetok** [comment]
@jhasse well, using wine isn't really a solution.
